src/views/Dashboard/Dashboard.tsx

```diff
@@ -9,13 +29,472 @@ import TopCards from "./components/TopCards";
+        reservesINDEX.push(tmpINDEX);
+
+        const tmpXSushi: TimeSeries = {
+          x: treasuryValues.blockNumbers[i],
+          y: 0,
+        };
+        reservesSushi.push(tmpXSushi);
+
+        const tmpUMA: TimeSeries = {
+          x: treasuryValues.blockNumbers[i],
+          y: 0,
+        };
+        reservesUMA.push(tmpUMA);
+      }
+    }
+
+    for (let i = 0; i < reservesHistory.length; i++) {
+      reserves.push({
+        x: reservesHistory[i].block,
+        y: reservesHistory[i].yUSD,
+      });
+      reservesDPI.push({
+        x: reservesHistory[i].block,
+        y: reservesHistory[i].DPI,
+      });
+      reservesETH.push({
+        x: reservesHistory[i].block,
+        y: reservesHistory[i].WETH,
+      });
+      reservesINDEXLP.push({
+        x: reservesHistory[i].block,
+        y: reservesHistory[i].INDEXLP,
+      });
+      reservesINDEX.push({
+        x: reservesHistory[i].block,
+        y: reservesHistory[i].INDEX,
+      });
+      reservesSushi.push({
+        x: reservesHistory[i].block,
+        y: reservesHistory[i].Sushi,
+      });
+      reservesUMA.push({
+        x: reservesHistory[i].block,
+        y: reservesHistory[i].UMA,
+      });
+    }
+
+    const series: SeriesInterface[] = [
+      {
+        name: "yUSD Reserves",
+        data: reserves ? reserves.slice(reserves.length - (defaultRange + 6)) : [],
+      },
+      {
+        name: "DPI Reserves",
+        data: reservesDPI ? reservesDPI.slice(reservesDPI.length - (defaultRange + 6)) : [],
+      },
+      {
+        name: "ETH Reserves",
+        data: reservesETH ? reservesETH.slice(reservesETH.length - (defaultRange + 6)) : [],
+      },
+      {
+        name: "Sushi Gains",
+        data: reservesSushi ? reservesSushi.slice(reservesSushi.length - (defaultRange + 6)) : [],
+      },
+      {
+        name: "INDEX Coop LP",
+        data: reservesINDEXLP ? reservesINDEXLP.slice(reservesINDEXLP.length - (defaultRange + 6)) : [],
+      },
+      {
+        name: "INDEX Coop Gains",
+        data: reservesINDEX ? reservesINDEX.slice(reservesINDEX.length - (defaultRange + 6)) : [],
+      },
+      {
+        name: "UMA Voting Token",
+        data: reservesUMA ? reservesUMA.slice(reservesUMA.length - (defaultRange + 6)) : [],
+      },
+    ];
+
+    let theme;
+    let labelColor;
+    let borderColor;
+    if (darkMode) {
+      theme = "dark";
+      labelColor = colors.grey[600];
+      borderColor = colors.grey[900];
+    } else {
+      theme = "light";
+      labelColor = colors.grey[600];
+      borderColor = colors.grey[600];
+    }
+
+    let options: OptionInterface = {
+      chart: {
+        background: "#ffffff00",
+        type: "area",
+        height: 350,
+        stacked: true,
+      },
+      stroke: {
+        curve: "stepline",
+      },
+      dataLabels: {
+        enabled: false,
+      },
+      fill: {
+        colors: ["#C60C4D", "#8150E6", "#4777e0", "#D16C00", "#838bfc", "#FFB900", "#2a9d8f"],
```

Comment:
for the colors you can do some short logic to pull it from the assets object coming from useTreasury, while setting it there